### PR TITLE
Add NewLocalShall function to get default local shell

### PR DESCRIPTION
**Pull request by magik6k**

### CODE REVIEW

File: shell.go [hunk 1]
```diff
@@ -20,11 +22,42 @@ import (
 	tar "github.com/whyrusleeping/tar-utils"
 )
 
+const (
+	DefaultPathName = ".ipfs"
```
Comment (lgierth) [inline shell.go]: Maybe use github.com/mitchellh/go-homedir instead and make this `DefaultIpfsPath = "~/.ipfs"`. That might simplify NewLocalShell a bit.

File: shell.go [hunk 2]
```diff
@@ -20,11 +22,42 @@ import (
...
+	DefaultApiFile  = "api"
```
Comment (lgierth) [inline shell.go]: This isn't really a default since it's not overridable :) But definitely something to have in a const.

File: shell.go [hunk 3]
```diff
@@ -20,11 +22,42 @@ import (
...
+		baseDir = path.Join(u.HomeDir, DefaultPathName)
+	}
+
+	apiFile := path.Join(baseDir, DefaultApiFile)
+
+	if _, err := os.Stat(apiFile); err != nil {
+		return nil
+	}
+
+	api, err := ioutil.ReadFile(apiFile)
```
Comment (lgierth) [inline shell.go]: Maybe better to strip whitespace from the end of the file content. Lots of editors add a trailing newline, and `echo` does too.

File: shell.go (hunk 2 above)
Comment (magik6k) [inline shell.go]: Those were taken from https://github.com/ipfs/go-ipfs/blob/master/repo/config/config.go#L34, decided to leave for consistency

**Comment by magik6k** [comment]
Circle needs setting IPFS_PATH for the tests to pass.

File: shell.go (hunk 3 above)
Comment (Kubuxu) [inline shell.go]: TrimSpace in line 59.

**Pull request merged**